File: BOJ/10448_1.cpp
// 유레카 이론
// https://www.acmicpc.net/problem/10448
#include <iostream>
#include <vector>
using namespace std;
vector<int> table;
int N;
bool ureka(int total)
{
    for (int i = 0; i < table.size(); i++)
        for (int j = 0; j < table.size(); j++)
            for (int k = 0; k < table.size(); k++)
                if (table[i] + table[j] + table[k] == total)
                    return true;
    return false;
}
int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    // 입력
    int tc;
    cin >> tc;
    // 삼각수 미리 계산
    int n = 1;
    while ((n) * (n + 1) / 2 < 1000)
    {
        table.push_back((n) * (n + 1) / 2);
        n++;
    }
    // 테스트
    for (int ts = 1; ts <= tc; ts++)
    {
        cin >> N;
        cout << ureka(N) << '\n';
    }
    return 0;
}

File: BOJ/10448_1.py
'''
    유레카 이론
    https://www.acmicpc.net/problem/10448
'''
from sys import stdin


def ureka(num):
    for i in range(len(T)):
        for j in range(len(T)):
            for k in range(len(T)):
                if num == T[i] + T[j] + T[k]:
                    return True
    return False


TC = int(stdin.readline())
n = 1
T = []
while (n) * (n+1) / 2 < 1000:
    num = (n) * (n+1) / 2
    T.append(num)
    n += 1
for ts in range(TC):
    N = int(stdin.readline())
    print(1 if ureka(N) else 0)
